Exercise_2.py

```diff
@@ -1,24 +1,48 @@
-
 class Node:
     def __init__(self, data):
-       self.data = data
-       self.next = None
- 
+        self.data = data
+        self.next = None
+
 class Stack:
     def __init__(self):
-        
+        self.head = None
+
+    # Time Complexity: O(N)
+    # Space Complexity: O(1)
     def push(self, data):
-        
+        new_node = Node(data)
```

Comment:
As you implementing Stack by using LinkedList, Push, pop should be done in O(1) right, So add/remove new node at head so it will be resulted in stack

Reply:
Oh Yess, that makes much more sense.
I over complicated things haha. 

This is how it should be, right? :

# O(1) TC and O(1) SC.
def push(self, data):
    new_node = Node(data)
    if self.head is None:
        self.head = new_node
    else:
        new_node.next = self.head
        self.head = new_node

# O(1) TC and O(1) SC.
def pop(self):
    if self.head is None:
        return None
    else:
        temp = self.head.data
        self.head = self.head.next
        return temp

Thanks for the feedback Jeel!

Best, 
Manan